clang/include/clang/AST/AbstractSet.h

```diff
@@ -78,7 +103,15 @@ namespace clang {
     llvm::DenseMap<PreorderAST *, const AbstractSet *> PreorderASTAbstractSetMap;
 
   public:
-    AbstractSetManager(Sema &S) : S(S) {}
+    AbstractSetManager(Sema &S, Sema::VarDeclUsage &VarUses) :
+      S(S), VarUses(VarUses) {}
+
+    ~AbstractSetManager() {
+      for (const auto &P : SortedPreorderASTs) {
+        P->Cleanup();
+        delete P;

```

Comment:
1) Ideally, lines 110 to 112 (i.e. invoking `Cleanup` on all `PreorderASTs`) should be moved to the function `Clear` in the `AbstractSetManager` class, and `Clear` should be called here. With this change, both the destructor and the function `Clear` will behave in a consistent manner. In the future, if the `Clear` method is called in more places, we do not want to leave leaked memory behind.
2) In the `Clear` method (and in the destructor), we also need to delete the AbstractSets, which are values in the `PreorderASTAbstractSetMap` before clearing the map.